src/spdx/parser/rdf/snippet_parser.py

```diff
@@ -0,0 +1,112 @@
+    copyright_text = parse_literal_or_no_assertion_or_none(logger, graph, snippet_node, SPDX_NAMESPACE.copyrightText)
+    comment = parse_literal(logger, graph, snippet_node, RDFS.comment)
+    name = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.name)
+    attribution_texts = []
+    for (_, _, attribution_text_literal) in graph.triples((snippet_node, SPDX_NAMESPACE.attributionText, None)):
+        attribution_texts.append(attribution_text_literal.toPython())
+
+    raise_parsing_error_if_logger_has_messages(logger, "Snippet")
+    snippet = construct_or_raise_parsing_error(Snippet,
+                                               dict(spdx_id=spdx_id, file_spdx_id=file_spdx_id, byte_range=byte_range,
+                                                    line_range=line_range, license_concluded=license_concluded,
+                                                    license_info_in_snippet=license_info_in_snippet,
+                                                    license_comment=license_comment,
+                                                    copyright_text=copyright_text, comment=comment, name=name,
+                                                    attribution_texts=attribution_texts))
+    return snippet
+
+
+def set_range_or_log_error(
+    byte_range: Optional[Tuple[int, int]], line_range: Optional[Tuple[int, int]], logger: Logger,
+    parsed_range: Dict[str, Tuple[int, int]]) -> Tuple[Optional[Tuple[int, int]], Optional[Tuple[int, int]]]:
+    if not parsed_range:
+        return byte_range, line_range
+    if "ByteOffsetPointer" in parsed_range.keys() and not byte_range:
+        byte_range = parsed_range["ByteOffsetPointer"]
+    elif "ByteOffsetPointer" in parsed_range.keys() and byte_range:
+        logger.append("Multiple ByteOffsetPointer found.")
+    elif "LineCharPointer" in parsed_range.keys() and not line_range:
+        line_range = parsed_range["LineCharPointer"]
+    elif "LineCharPointer" in parsed_range.keys() and line_range:
+        logger.append("Multiple LineCharPointer found.")
+    return byte_range, line_range
+
+
+def parse_ranges(start_end_pointer: URIRef, graph: Graph) -> Dict[str, Tuple[int, int]]:
+    range_values = dict()
+    for pointer_property in [POINTER_NAMESPACE.startPointer, POINTER_NAMESPACE.endPointer]:
+        try:
+            pointer_node = graph.value(start_end_pointer, pointer_property, any=False)
+        except UniquenessError:
+            raise SPDXParsingError([f"Multiple values for {pointer_property.fragment}"])
+        if not pointer_node:
+            raise SPDXParsingError([f"Couldn't find pointer of type {pointer_property.fragment}."])
+
+        pointer_type = graph.value(pointer_node, RDF.type)
+        range_values[pointer_property] = parse_range_value(graph, pointer_node, POINTER_MATCHING[pointer_type])
+
+    return {str(pointer_type.fragment): (
+        range_values[POINTER_NAMESPACE.startPointer], range_values[POINTER_NAMESPACE.endPointer])}
```

Comment:
`pointer_type` may be different for `startPointer` and `endPointer`, but only the type of `endPointer` will be returned. As this would mean an invalid SPDX document, we should check that the pointer types match and raise an error otherwise

Reply:
You are right, these ranges make me crazy :exploding_head: